lib/node.js

```diff
@@ -175,13 +175,15 @@ Node.prototype.delete = function (key, distinct) {
 };
 
 Node.prototype._deleteRecursive = function (key, distinct, index) {
-  if (index === key.length || this.leaf) {
+  if (this.leaf) {
     // If the index matches the key length, we are at the node to be deleted.
     // Remove values associated with the key, where the `distinct` value also matches.
-    this.values = this.values.filter((item) => (item.key !== key && item.distinct !== distinct));
+    this.values = this.values.filter(
+      (item) => item.key !== key && item.distinct !== distinct
+    );
 
     // If the node becomes a leaf node, update its status.
-    this.leaf = this.values.length > 0;
+    // this.leaf = this.values.length > 0;
```

Comment:
To confirm - I don't think we actually care about changing leaf to non-leaf when deleting. It'll _**maybe**_ save a recursive loop when querying, but isn't strictly necessary.